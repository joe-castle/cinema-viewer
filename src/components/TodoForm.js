import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Input = styled.input`
  border: 1px solid #ccc;
  font-size: 25px;
  padding: 15px;
  width: 100%;

  &:hover {
    border-color: rgba(0, 0, 0, .7);
  }
`

export default class TodoForm extends React.Component {
  static propTypes = {
    addTodo: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)
    this.state = { todoText: '' }
  }

  render () {
    return (
      <form
        onSubmit={(ev) => {
          ev.preventDefault()
          if (this.state.todoText !== '') {
            this.props.addTodo(this.state.todoText)
            this.setState({ todoText: '' })
          }
        }}
      >
        <Input
          placeholder='Enter new todo...'
          value={this.state.todoText}
          onChange={(e) => {
            this.setState({ todoText: e.target.value })
          }}
        />
      </form>
    )
  }
}
