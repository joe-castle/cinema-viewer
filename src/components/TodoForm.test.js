import React from 'react'
import { shallow, mount } from 'enzyme'

import TodoForm from './TodoForm'

describe('TodoForm', () => {
  const addTodo = jest.fn()

  beforeEach(() => {
    addTodo.mockClear()
  })

  test('Should render', () => {
    const wrapper = mount(<TodoForm addTodo={addTodo} />)

    expect(addTodo).toBeCalledTimes(0)
    expect(wrapper).toMatchSnapshot()
  })

  test('Should call addTodo when form submitted with value in input', () => {
    const wrapper = mount(<TodoForm addTodo={addTodo} />)

    wrapper.setState({ todoText: 'todo' })

    expect(wrapper.find('input').prop('value')).toBe('todo')

    wrapper.find('form').simulate('submit', { preventDefault () {} })

    expect(addTodo).toHaveBeenCalledTimes(1)
    expect(addTodo).toHaveBeenCalledWith('todo')
    expect(wrapper.find('input').prop('value')).toBe('')
  })

  test('Should not call addTodo if form submitted with no value in input', () => {
    const wrapper = shallow(<TodoForm addTodo={addTodo} />)

    wrapper.find('form').simulate('submit', { preventDefault () {} })

    expect(addTodo).toHaveBeenCalledTimes(0)
  })
})
