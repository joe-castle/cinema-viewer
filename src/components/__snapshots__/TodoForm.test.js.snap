// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm Should render 1`] = `
.c0 {
  border: 1px solid #ccc;
  font-size: 25px;
  padding: 15px;
  width: 100%;
}

.c0:hover {
  border-color: rgba(0,0,0,.7);
}

<TodoForm
  addTodo={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <TodoForm__Input
      onChange={[Function]}
      placeholder="Enter new todo..."
      value=""
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "TodoForm__Input-p2gkw7-0",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "border:1px solid #ccc;font-size:25px;padding:15px;width:100%;&:hover{border-color:rgba(0,0,0,.7);}",
              ],
            },
            "displayName": "TodoForm__Input",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "TodoForm__Input-p2gkw7-0",
            "target": "input",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onChange={[Function]}
        placeholder="Enter new todo..."
        value=""
      >
        <input
          className="c0"
          onChange={[Function]}
          placeholder="Enter new todo..."
          value=""
        />
      </StyledComponent>
    </TodoForm__Input>
  </form>
</TodoForm>
`;
